book/build-service.md

```diff
@@ -0,0 +1,142 @@
+
+## Controllers
+
+The Build Service contains these controllers:
+- Component Build Controller
+  - Used for PipelineRuns creation.
+- PaC PipelineRun Pruner Controller
+  - Deletes PipelineRuns managed by Pipelines As Code for Components that were deleted.
+- Git Tekton Resources Renovater
+  - Provides updates for the .tekton directory in the user repository which has been created by Component Build Controller in Custom Mode.
+
+### Component Build Controller
+
+Component Build Controller is managed by Component CR changes. It's using Component CR annotations and `.status.devfile` for selecting a mode and configuration of the PipelineRuns.
+
+#### Modes
+
+Component Build Controller is working in two modes:
+- Default Mode
+  - Mode where PipelineRuns are created directly by the Build Service.
+- Custom Mode
+  - Integrates with Pipelines-As-Code, the Component Build Controller creates PipelineRuns definitions in the user code repository.
+  - The prerequisite is to have installed GitHub App which is used by the Build Server in the user's repository.
+
+Default mode:
+1. Wait until the component has `.status.devfile` set
+1. Get data from `.status.devfile` and use them for creating PipelineRun
+1. Set annotation on the component - `appstudio.openshift.io/component-initial-build: processed`
+
+Delete annotation `appstudio.openshift.io/component-initial-build` on Component to retrigger PipelineRun in Default mode.
+
+For the selection of Custom mode the Component CR must have set annotation `appstudio.openshift.io/pac-provision` to `request`.
+
+Custom Mode:
+1. Wait until the component has `.status.devfile` set
+1. Get data from `.status.devfile`, use them for creating `.tekton` folder in the user's repository and create Repository CR so that Pipelines-as-Code is configured to monitor the user's repository.
+1. Change the value of `appstudio.openshift.io/pac-provision` annotation to `done` in case the previous step was successful, in case of an issue the annotation is set to `error` and annotation `appstudio.openshift.io/pac-provision-error` with more details about the error is added.
+
+#### PipelineRun selection
+
+The Build Service owns [BuildPipelineSelector CRD](https://redhat-appstudio.github.io/book/ref/build-service.html#buildpipelineselector) which defines which PipelineRun to select for the Component CR. By default global BuildPipelineSelector `build-pipeline-selector` in `build-service` namespace is used. BuildPipelineSelector CR contains selectors, the first matching selector is used for the Component CR. The list of selectors can be extended by creating BuildPipelineSelector CR named the same as Application CR in the user's namespace so that will be applied to Components CRs under the Application CR. The list of selectors can be also extended for the whole user namespace by creating BuildPipelineSelector CR named `build-pipeline-selector` in the user's namespace.
```

Comment:
`The list of selectors can be extended by creating BuildPipelineSelector CR named the same as Application CR in the user's namespace so that will be applied to Components CRs under the Application CR.`
seems too long, personally I would split it into two sentences:
`The list of selectors can be extended by creating BuildPipelineSelector CR with the same name as the Application CR in the user's namespace. This will ensure that it is applied to Component CRs under the corresponding Application CR.`